### PR TITLE
[MLU] add all kernel

### PR DESCRIPTION
**Pull request by PeiyuLau**
add all kernel

### CODE REVIEW

File: backends/mlu/tests/unittests/test_reduce_all_op_mlu.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        self.attrs = {"reduce_all": True}
+
+    def set_mlu(self):
+        self.__class__.use_custom_device = True
+
+    def test_check_output(self):
+        self.check_output_with_place(self.place)
+
+
+# class TestAllOp_ZeroDim(OpTest):
```
Comment (ShawnNew) [inline backends/mlu/tests/unittests/test_reduce_all_op_mlu.py]: delete if not needed

File: backends/mlu/kernels/reduce_all_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (ShawnNew) [inline backends/mlu/kernels/reduce_all_kernel.cc]: 2022->2024

File: backends/mlu/kernels/reduce_all_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  dev_ctx.template Alloc<T>(out);
+  bool reduce_all =
+      dims.size() == 0 || static_cast<int>(dims.size()) == x.dims().size();
+  MLUReduceOp<T>(dev_ctx, x, dims, keep_dim, reduce_all, "reduce_all", out);
+}
+
+}  // namespace custom_kernel
+
+PD_REGISTER_PLUGIN_KERNEL(
+    all, mlu, ALL_LAYOUT, custom_kernel::AllKernel, bool) {}
```
Comment (ShawnNew) [inline backends/mlu/kernels/reduce_all_kernel.cc]: other dtypes?

**Review by ShawnNew — CHANGES_REQUESTED** [review]

File: backends/mlu/tests/unittests/test_reduce_all_op_mlu.py (hunk 1 above)
Comment (PeiyuLau) [reply backends/mlu/tests/unittests/test_reduce_all_op_mlu.py]: modified

File: backends/mlu/kernels/reduce_all_kernel.cc (hunk 2 above)
Comment (PeiyuLau) [reply backends/mlu/kernels/reduce_all_kernel.cc]: modified

File: backends/mlu/kernels/reduce_all_kernel.cc (hunk 3 above)
Comment (PeiyuLau) [reply backends/mlu/kernels/reduce_all_kernel.cc]: add float , int and int64_t dtype.

**Review by ShawnNew — APPROVED** [review]
LGTM

**Pull request merged**